python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -0,0 +1,262 @@
+               calculating accuracy.
+        :param direction: (optional) A string that indicates the higher/lower
+               better for the metric, "min" for the lower the better and "max" for the
+               higher the better. Default value is "max".
+        :param thread_num: (optional) a int represents how many threads(cores) is needed for
+               inference.
+        :param logging: whether to log detailed information of model conversion.
+               Default: False.
+        :param latency_sample_num: (optional) a int represents the number of repetitions
+               to calculate the average latency. The default value is 100.
+        :param includes: (optional) a list of acceleration methods that will be included in the
+               search. Default to None meaning including all available methods. "original" method
+               will be automatically add to includes.
+        :param excludes: (optional) a list of acceleration methods that will be excluded from the
+               search. "original" will be ignored in the excludes.
+        '''
+        # check if model is a nn.Module or inherited from a nn.Module
+        invalidInputError(isinstance(model, Model), "model should be a Keras Model.")
+        invalidInputError(direction in ['min', 'max'],
+                          "Only support direction 'min', 'max'.")
+
+        if not isinstance(model, NanoModel):
+            # turn model into NanoModel to obtain trace and quantize method
+            model = NanoModel(inputs=model.inputs, outputs=model.outputs)
+
+        # get the available methods whose dep is met
+        available_dict: Dict =\
+            available_acceleration_combination(excludes=excludes,
+                                               includes=includes,
+                                               full_methods=self.ALL_INFERENCE_ACCELERATION_METHOD)
+
+        self._direction: str = direction  # save direction as attr
+        # record whether calculate accuracy in optimize by this attr
+        if validation_data is None or metric is None:
+            self._calculate_accuracy = False
+        else:
+            # test whether accuracy calculation works later
+            # make sure dataset don't have batch
+            batched_validation_data = validation_data.batch(batch_size)
+            self._calculate_accuracy = True
+
+        if os.getenv('OMP_NUM_THREADS') is not None:
+            default_threads: int = int(os.getenv('OMP_NUM_THREADS'))  # type: ignore
+        else:
+            # TODO: how to get and control thread num in tf?
+            default_threads = None  # type: ignore
+        thread_num = default_threads if thread_num is None else int(thread_num)  # type: ignore
+
+        result_map: Dict[str, Dict] = {}
+
+        batched_training_data = training_data.batch(batch_size)
+        input_sample = next(iter(batched_training_data))
+        # TODO: how to obtain input from output of training_data
```

Comment:
Maybe we can use the same input requirement as `tf.keras.model.fit` (https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit), batched or unbatched version.

Reply:
This maybe a way to obtain more accurate input. But it will occur different input mode from torch.InferenceOptimizer.
How do you think about this ? @TheaperDeng @MeouSker77

Reply:
I think it's better to be consistent with the common usage of tensorflow users (i.e. be same as `tf.keras.model.fit`). But the inc, openvino and onnx quantization all requires a dataset which produces both `x` and `y`, so we have to do some changes if we use the same input requirements as `tf.keras.model.fit`. But I think these changes are acceptable.

Reply:
also agree to change the parameter to x, y here. We may do it in later PR.

Reply:
well, will change the parameter to x, y in next PR.